fbw-a32nx/src/systems/instruments/src/EFB/Apis/Navigraph/Components/Authentication.tsx

```diff
@@ -7,9 +7,42 @@ import React, { useEffect, useState } from 'react';
 import { CloudArrowDown, ShieldLock } from 'react-bootstrap-icons';
 import { toast } from 'react-toastify';
 import QRCode from 'qrcode.react';
-import { usePersistentProperty } from '@flybywiresim/fbw-sdk';
+import { NavigraphClient, NavigraphSubscriptionStatus, usePersistentProperty } from '@flybywiresim/fbw-sdk';
+import { useHistory } from 'react-router-dom';
 import { t } from '../../../translation';
-import NavigraphClient, { useNavigraph } from '../Navigraph';
+import { useNavigraph } from '../Navigraph';
+
+export type NavigraphAuthInfo = {
+    loggedIn: false,
+} | {
+    loggedIn: true,
+
+    username: string,
+
+    subscriptionStatus: NavigraphSubscriptionStatus,
+}
+
+export const useNavigraphAuthInfo = (): NavigraphAuthInfo => {
+    const navigraph = useNavigraph();
+
+    const [tokenAvail, setTokenAvail] = useState(false);
+    const [subscriptionStatus, setSubscriptionStatus] = useState<NavigraphSubscriptionStatus>(NavigraphSubscriptionStatus.None);
+
+    useInterval(() => {
+        if ((tokenAvail !== navigraph.hasToken) && navigraph.hasToken) {
+            navigraph.subscriptionStatus().then(setSubscriptionStatus);
```

Comment:
Is unhandled promise rejection if there's a network error okay here? It will leave you with the sub status still as none so I guess so?

Reply:
We could do a "Unknown" state maybe?

Reply:
Done